### ISSUE TITLE
HACS Home Assistant Integration: API gives very inaccurate weather data

### ISSUE THREAD

**Issue by Jm-Fox** [Pirate-Weather/pirate-weather-ha#220]
### Describe the bug

First all, thanks for providing this tool and the free tier. I look forward to using it further.

Second, I don't expect perfect weather forecasts / current temperatures / etc. But for me to rely on it in home assistant, I do need values to be "good enough."

My (long form) issue is as follows:
 * Right now the API used by Home Assistant is giving significantly different data than the Merry Sky website.
 * Time: ~12:30 PM CST April 21, 2024
 * Location: 32.952199718765016, -96.66548605790446 (approximate- a mile or so from my actual location)
 * Merry Sky current reported temperature: 58 degrees Fahrenheit (65% humidity)
 * API current reported temperature: 25 degrees Fahrenheit (88% humidity)
   * 3, 6, 12, and 24 hour forecasts are similarly chilly (20-30 degrees under forecasts of other services)
 * weather.com current reported temperature: 62 degrees Fahrenheit (40% humidity)

The difference between Merry Sky and weather.com I think is easily explained by documented caching in Merry Sky (issues Pirate-Weather/pirateweather#30, Pirate-Weather/pirateweather#156). However, the API being off by 30 degrees and 2x humidity is wild. At first I questioned user error: bad coordinates, but after triple checking everything I'm still stuck. I apologize if this has been discussed elsewhere.

To my knowledge, I have been using the production API endpoint. If I can assist in verification by tweaking settings, I am able to (given instructions / links to documentation).

### Expected behavior

Pirate API in Home Assistant to report approximate temperatures- within a few degrees of other services. Same for humidity.

### Actual behavior

Pirate API in Home Assistant is reporting wildly different temperatures / humidity than other services, including Merry Sky, which should be operating on the same data (if a bit outdated due to caching forecasts).

### API Endpoint

Production

### Location

North Shiloh Road, Garland, Dallas County

### Other details

I confess I did not fully read through the API documentation before opening the issue. Since I'm relying on the integration to make the API calls and am not making them personally, I assumed there'd be little value in my review of the api docs. I apologize if this should be an issue on the [HA repository](https://github.com/Pirate-Weather/pirate-weather-ha).

### Troubleshooting steps

- [X] I have searched this repository and [Home Assistant Repository](https://github.com/alexander0042/pirate-weather-ha) to see if the issue has already been reported.
- [X] I have read through the [API documentation](https://pirateweather.net/en/latest/API/) before opening this issue.
- [X] I have written an informative title.

**Comment by Jm-Fox** [comment]
I'm new to this- how do I get the logs you need?

My best guess was:
1. Enable debug logging from the Pirate Weather integration configuration page (home assistant -> /config/integrations/integration/pirateweather)
2. Reload the integration
3. Review entries in the Logbook tab of Home Assistant (see images below; it's currently ~63 degrees Fahrenheit outside, so the issue is still reproducing).
   * It's interesting that the daily forecast includes Sunday, when local time is ~10:45 am Monday. The hourly forecast at least has the correct timestamps on it.

![image](https://github.com/Pirate-Weather/pirate-weather-ha/assets/6495375/6827ad7d-17e8-4bb0-a6fb-73936e465ef3)
![image](https://github.com/Pirate-Weather/pirate-weather-ha/assets/6495375/13f50054-8039-41f2-b4b9-f8494cd2ec47)

**Comment by alexander0042** [comment]
Great question! The logs are visible in the WebUI here: https://homeassistant.local/config/logs, and you're looking for any line that starts with Pirate Weater- just search for it and see if anything comes up. 

What's weird is looking at that it's not looking like an update issue, but rather something related to the data. Did you ever change the location it's requesting, since there's a limitation where you can't edit the location after it's been created.

**Comment by Jm-Fox** [comment]
That's it: I did change the location. I had goofed the longitude on my first attempt, missing the negative sign.

I confirmed Merry Sky gave the same data if I give it the positive longitude, then deleted and recreated the home assistant integration. All the numbers line up now.

For good measure I also reviewed the /config/logs URL and didn't see anything unusual (a timeout and a 502, but I know those things happen from time to time).

Thank you for your assistance!
